### PR TITLE
feat: refactor accessibility features to reflect state on host

### LINKED ISSUE #1611: [A11y] more aria props for sbb-card ♿
We need for example the aria-expanded attribute to show Screenreaders that something happend in the dom/screen.

It seems like that you can't use this attribute on the Host element. So I would assume that it have to forwarded directly to the button element with a prop.

### PR DESCRIPTION
**Pull request by kyubisation**
Closes #1611 

This PR refactors our interactive elements to have the state and focus on the host element. This allows consumers to create aria references on the and to the host, which was previously not possible.

BREAKING CHANGE: Almost all accessibility-label attributes were removed in favor of just using aria-label (exceptions are `<sbb-alert>`, `<sbb-dialog>`, `<sbb-logo>`, `<sbb-navigation>`, `<sbb-navigation-section>`, `<sbb-signet>` and `<sbb-train>`).

### CODE REVIEW

File: src/global/interfaces/accessibility-properties.ts [hunk 1]
```diff
@@ -1,21 +0,0 @@
-/**
- * The accessibility interface is used to enforce all accessibility labels to be implemented at once.
- */
-export interface AccessibilityProperties {
```
Comment (jeripeierSBB) [inline src/global/interfaces/accessibility-properties.ts:4]: I'd keep the interface for the remaining usages, so we can still maintain a consistent naming.

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Thanks a lot for getting into it!!!

Some general points:
- Documentation of sbb-button, sbb-checkbox, sbb-toggle-check accessibility-properties should be updated
- Stories of sbb-checkbox, toggle-option should be adapted
- Can you recheck also all other documentation (readme, stories)? I did not check all. In general, I think it would help to have also aria-labels as controls. So our examples are complete.

File: src/components/sbb-card-badge/sbb-card-badge.tsx [hunk 2]
```diff
@@ -63,14 +52,12 @@ export class SbbCardBadge implements ComponentInterface, AccessibilityProperties
...
     return (
       <span
-        aria-label={this.accessibilityLabel}
         class={`card-badge
           ${appearanceClass}
           ${sizeClass}`}
         dir={currentWritingMode}
         // eslint-disable-next-line jsx-a11y/aria-role
         role="text"
-        title={this.accessibilityLabel}
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/components/sbb-card-badge/sbb-card-badge.tsx:73]: Yes, a bit strange. I just removed it, as I was removing the legacy accessibility-label occurrences.

File: src/components/sbb-dialog/sbb-dialog.tsx [hunk 3]
```diff
@@ -41,7 +40,7 @@ type SbbDialogState = 'closed' | 'opening' | 'opened' | 'closing';
   styleUrl: 'sbb-dialog.scss',
   tag: 'sbb-dialog',
 })
-export class SbbDialog implements ComponentInterface, AccessibilityProperties {
+export class SbbDialog implements ComponentInterface {
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/components/sbb-dialog/sbb-dialog.tsx:43]: Discussed via Teams; We will remove it.

File: src/components/sbb-tag/sbb-tag.tsx [hunk 4]
```diff
@@ -37,7 +35,10 @@ import { TagStateChange } from './sbb-tag.custom';
   styleUrl: 'sbb-tag.scss',
   tag: 'sbb-tag',
 })
-export class SbbTag implements ComponentInterface, AccessibilityProperties {
+export class SbbTag implements ComponentInterface {
+  /** The name attribute to use for the button. */
+  @Prop() public name: string | undefined;
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/components/sbb-tag/sbb-tag.tsx:40]: Good point, it does not. But it should be a property of form controls, however it should probably be reflected.

**Comment by federicoisepponfincons** [comment]
**Chrome + JAWS**

Disclaimer: probably many of these comments are just nitpicking
Notation: text included in *stars* report what the screen reader actually reads

SBB alert - works fine, close button is correctly read

SBB button - works fine, disable button is read as *button button inavailable* with the first "button" that is the name of the button label (that's acceptable in my opinion). The "icon only" variant is read as *button name button* (in this case it should read the icon name?)

SBB card - works fine. "Sbb card with badge" variant: the badge is not read by JAWS. Other variants are ok

SBB dialog - opening button ok. Back button is read as *unlabeled 0 button* and close button is read as *unlabeled 2 button*. Rest of the dialog works fine. Variants are ok

SBB header action - ok

SBB link - ok

SBB menu action - links and buttons work fine. JAWS reads *link details one hundred and 99*, probably because there is nothing between the label "details 1" and the amount element, and the reader takes them as a single number (199) - corner case because usually those labels don't end with numbers.

SBB navigation action - ok

SBB navigation section - hamburger menu button is read as *unlabeled zero button*. With the navigation open, the close button is positioned before the action elements, so it is read before them. The close button is also read as *unlabeled zero button*, so it needs to be more specific.

SBB navigation - same as navigation section

SBB radio button group - JAWS the read radio buttons one by one, works fine. Disabled radio button are read as *value one radio button not checked unavailable one of four*, that's good

SBB slider - slider is read as *slider left right slider 40 min 0 max 100* where 40 is the current value, and with tab I can change the value, I thinks this work very well. 

SBB tag - works fine

SBB teaser hero - ok, the link is correctly read

SBB teaser - JAWS here works perfectly. The focus tab outline need to be visually fixed

SBB time input - JAWS reads *HH colon MM* at first (input hidden placeholder), then sometimes it doesn't read the value, sometimes it reads *twelve o clock*. With a label that explain the input meaning, I think this could be understandable by the user.

SBB toggle option - ok

**Comment by federicoisepponfincons** [comment]
**Chrome + NVDA**

Disclaimer and notation are the same as Chrome + JAWS :)


SBB alert - tab navigation is ok. Close button is correctly read as *close message button*. On mouse hover (visually impaired user) the link is correctly read, but the close button is read only as *button*.

SBB button - Works fine. In the "icon only" variant, should it read the icon name? Because nothing else can be read here. "Disabled" variant: not reading the button, should it read the button even if it's disabled?

SBB card - title <sbb-title> is not read, only the content text is read

SBB dialog - opening the dialog is ok, reading of the title is ok, reading of the content is ok, back and close button are read as *blank* when you hover them, and are read as *button* on tab focus, so they need a more specific label. Link, cancel and confirm buttons are ok. All other variants are ok

SBB header action - action and links both work fine

SBB link - links and buttons both work fine

SBB menu action - links and buttons work fine. Same "problem" as JAWS with numbers. Disabled links are not read

SBB navigation action - ok

SBB navigation section - close button is read simply as *button* without other specifications. Hamburger menu button is ok.Navigation and its elements are ok

SBB navigation - same as navigation section

SBB radio button group - using the tab to navigate the page, it only select the currently checked radio button, then the reader does not read all the radio options but only the one that is selected, so the user could not know that there are multiple options. you can select a different radio button with arrow keys, but I would expect that with the tab you highlight the whole radio button group and have it read completely

SBB slider - it reads *slider 30* where 30 is the current value. shouldn't we provide a more understandable name to the slider itself? or do we rely on the label that should be added by the user? Currently the "sbb slider in form field" variant provides a label, NVDA does not read the label but only the slider value. I tried placing a label outside the sbb-form-field and this is read by the screen reader. Disabled or readonly slider aren't read at all.

SBB tag - working fine, but numbers are read as a single one, eg: label 2 123 (where 123 is the amount element) is read *label two thousands one hundred and twenty three* - corner case because usually those labels don't end with numbers

SBB teaser hero - ok

SBB teaser - NVDA reads it correctly. The focus tab outline need to be visually fixed

SBB time input - NVDA reads *HH MM twelve o clock* where HH MM is the placeholder of the input field. I did a little research about time reading, and there is no general rule about how it should be spelled; first answer in: https://stackoverflow.com/questions/75322134/nvda-screen-reader-ignores-time-element

SBB toggle option - works fine

**Comment by federicoisepponfincons** [comment]
**Firefox + JAWS**

navigation with arrow keys seems to be broken, it seems that only tab navigation works atm

**Comment by federicoisepponfincons** [comment]
**Firefox + NVDA**

SBB alert - all is working fine with arrow keys navigation. On mouse hover the close button is read only as *button*.

SBB button - NVDA just reads the button label and that could be acceptable. The "icon only" variant hasn't a label, so it is just read as *button*. "Disabled" variant: button is read on hover, but it's not specified that is disabled.

SBB card - with tab navigation it works fine. On hover, only the content text is read (as a link text)

SBB dialog - everything is working fine, except the "back" and "close" button that are read as *button* on arrow keys navigation, and as *blank* on mouse over. As chrome they need a more specific label. All other variants are ok

SBB header action - ok

SBB link - ok

SBB menu action - links and buttons work fine. Same "problem" as JAWS with numbers. Disabled links are not read

SBB navigation action - ok

SBB navigation section - Hamburger menu button is ok. Close button is read simply as *button* without other specifications. Navigation and its elements are ok

SBB navigation - same as navigation section

SBB radio button group - same as chrome + NVDA

SBB slider - same as chrome + NVDA

SBB tag - ok

SBB teaser hero - ok

SBB teaser - same as chrome + NVDA, focus outline has to be fixed

SBB time input - same as chrome + NVDA

SBB toggle option - ok

**Pull request merged**